cmd/contour/cli.go

```diff
@@ -15,41 +15,96 @@ package main
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/ioutil"
+	"log"
 	"os"
 
-	"github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	"github.com/gogo/protobuf/proto"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
+// Client holds the details for the cli client to connect to.
 type Client struct {
 	ContourAddr string
+	CAFile      string
+	ClientCert  string
+	ClientKey   string
 }
 
 func (c *Client) dial() *grpc.ClientConn {
-	conn, err := grpc.Dial(c.ContourAddr, grpc.WithInsecure())
-	check(err)
+
+	var conn *grpc.ClientConn
```

Comment:
You can eliminate these local declarations as follows

Reply:
Just delete `var err error`, each line below that does something like `conn, err := ...` has declared a new err inside the nested block, effective shadowing it.